hashes/src/sha256t/mod.rs

```diff
@@ -135,13 +137,6 @@ impl<T: Tag> core::hash::Hash for Hash<T> {
 
 crate::internal_macros::hash_trait_impls!(256, false, T: Tag);
 
-fn from_engine<T>(e: sha256::HashEngine) -> Hash<T>
-where
-    T: Tag,
-{
-    Hash::from_byte_array(sha256::Hash::from_engine(e).to_byte_array())
-}
-
```

Comment:
> `sha256t` module does not use that macro.

This is actually a problem, I believe. We are risking SSOT issues all over again. Why is this the case?

I don't think this should be removed but rather the macro should be called or a good reason given why the macro cannot be used.

Reply:
We should change every other instance of the macro to stop using a standalone `from_engine` method. But we can do that in a followup PR. (And maybe once we clean up the macro a bit we can use it in sha256t again.)

Reply:
OK, if we want to do that, and I think it does make sense, then this commit is fine. Previously I was thinking of putting the function back to allow the macro.